src/main/java/org/apache/maven/plugins/scripting/engine/JavaEngine.java

```diff
@@ -0,0 +1,327 @@
+                            "-d", bin.toAbsolutePath().toString(),
+                            java.toAbsolutePath().toString() )
+                    .toArray( String[]::new ) );
+            if ( run != 0 )
+            {
+                throw new IllegalArgumentException(
+                        "Can't compile the incoming script, here is the generated code: >\n" + source + "\n<\n" );
+            }
+            final URLClassLoader loader = new URLClassLoader(
+                    new URL[]{ bin.toUri().toURL() },
+                    Thread.currentThread().getContextClassLoader() );
+            final Class<? extends CompiledScript> loadClass =
+                    loader.loadClass( packageName + '.' + className ).asSubclass( CompiledScript.class );
+            return loadClass
+                    .getConstructor( ScriptEngine.class, URLClassLoader.class )
+                    .newInstance( this, loader );
+        }
+        catch ( Exception e )
+        {
+            throw new ScriptException( e );
+        }
+        finally
+        {
+            if ( tmpDir != null )
+            {
+                try
+                {
+                    Files.walkFileTree( tmpDir, new SimpleFileVisitor<Path>()
+                    {
+                        @Override
+                        public FileVisitResult visitFile( Path file, BasicFileAttributes attrs )
+                                throws IOException
+                        {
+                            Files.delete( file );
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult postVisitDirectory( Path dir, IOException exc )
+                                throws IOException
+                        {
+                            Files.delete( dir );
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                    } );
+                }
+                catch ( IOException e )
+                {
+                    // no-op: todo: throw an exception if not already thrown or add a suppressed
+                }
+            }
+        }
+    }
+
+    private String mavenClasspathPrefix()
+    {
+        final String home = System.getProperty( "maven.home" );
+        if ( home == null )
+        {
+            return "";
+        }
+        try ( Stream<Path> files = Files.list( Paths.get( home ).resolve( "lib" ) ) )
+        {
+            return files
+                    .filter( it ->
+                    {
+                        final String name = it.getFileName().toString();
+                        return name.startsWith( "maven-" );
+                    } )
+                    .map( Path::toString )
+                    .collect( joining( File.pathSeparator, "", File.pathSeparator ) );
+        }
+        catch ( IOException e )
+        {
+            return "";
+        }
```

Comment:
Please at least log the exception with a minor log level

Reply:
done